### DISCUSSION TITLE
createDerivedMaterial - custom attributes?

### DISCUSSION THREAD

**Discussion by mokargas** [protectwise/troika#298] [Q&A]
I'm using instanced uniforms mesh with createDerivedMaterial.

I have a custom instanced buffer attribute on the mesh (jus stores initial Positions), however I cant seem to access it in vertexDefs. Is this unsupported?

```javascript
  const customMaterial = useMemo(() => {
    const base = new MeshBasicMaterial({
      side: DoubleSide,
      transparent: true,
      toneMapped: false,
    });
    const vertexDefs = `
      uniform int uHoveredInstanceId;
      attribute vec3 initialPosition;

    `;
    const vertexTransform = `
      position = initialPosition;
  `;

    return createDerivedMaterial(base.clone(), {
      timeUniform: 'elapsed',
      uniforms: {
        uHoveredInstanceId: { value: -1 },
        time: { value: 0 },
      },
      vertexDefs,
      vertexTransform,
    });
  }, []);
```

**Comment by lojjic** [comment]
Hmm, I _feel_ like this should work, but chaining shader transforms can sometimes be fragile.

Do you get shader compile errors? Or does it just ignore the initialPosition?

**Reply by mokargas** [reply]
No compilation errors, it just ignores the initialPosition.

 I eventually abandoned using instanced uniforms mesh and used an approach that avoided using instanced uniforms for my use case, but would like it to work in future. I thought perhaps I was missing something fundamental, like needing to transform the position into worldspace but no go.
